pkg/backup/backupdest/backup_destination_test.go

```diff
@@ -397,4 +399,92 @@ func TestBackupRestoreResolveDestination(t *testing.T) {
 	}
 }
 
+func TestMixedVersionIncrementalSuffixChains(t *testing.T) {
+	defer leaktest.AfterTest(t)()
+	defer log.Scope(t).Close(t)
+
+	args := base.TestServerArgs{
+		DefaultTestTenant: base.TestControlsTenantsExplicitly,
+	}
+	args.Knobs.Server = &server.TestingKnobs{
+		ClusterVersionOverride:         clusterversion.V25_1.Version(),
+		DisableAutomaticVersionUpgrade: make(chan struct{}),
+	}
+	_, db, tmpdir, cleanup := backuptestutils.StartBackupRestoreTestCluster(
+		t, backuptestutils.SingleNode, backuptestutils.WithParams(base.TestClusterArgs{
+			ServerArgs: args,
+		}),
+	)
+	defer cleanup()
+	fmt.Println(tmpdir)
+
+	db.Exec(t, "CREATE TABLE foo (a int)")
+	db.Exec(t, "BACKUP INTO 'nodelocal://1/backup'")
+
+	db.Exec(t, "INSERT INTO foo VALUES (1)")
+	// Round to nearest millisecond timestamp to avoid precision shenanigans
+	// between MacOS and Unix.
+	timeBeforeUpgrade := hlc.Timestamp{WallTime: time.Now().UnixNano() / 1e3 * 1e3}
+	db.Exec(
+		t,
+		fmt.Sprintf(
+			"BACKUP INTO LATEST IN 'nodelocal://1/backup' AS OF SYSTEM TIME '%s'",
+			timeBeforeUpgrade.AsOfSystemTime(),
+		),
+	)
+
+	backups := db.QueryStr(
+		t,
+		"SELECT DISTINCT (start_time, end_time) FROM [SHOW BACKUP FROM LATEST IN 'nodelocal://1/backup']",
+	)
+	require.Len(t, backups, 2)
```

Comment:
nit: could you verify that the old dir structure is used before the migration? you could use a regex with:
```
SELECT path FROM [SHOW BACKUP FILES FROM LATEST IN $1 ] LIMIT 1
```

Reply:
‼️‼️ I totally forgot about that query, I've been wanting some way of validating the backup paths forever.